src/accel/FastSimulationOffload.cc

```diff
@@ -0,0 +1,88 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file accel/FastSimulationOffload.cc
+//---------------------------------------------------------------------------//
+#include "FastSimulationOffload.hh"
+
+#include "corecel/Assert.hh"
+
+#include "ExceptionConverter.hh"
+#include "LocalTransporter.hh"
+#include "SharedParams.hh"
+
+namespace celeritas
+{
+FastSimulationOffload::FastSimulationOffload(G4String const& name,
+                                             SharedParams const* params,
+                                             LocalTransporter* local)
+    : G4VFastSimulationModel(name), params_(params), transport_(local)
+{
+    CELER_VALIDATE(G4VERSION_NUMBER > 1110,
+                   << "the current version of Geant4 (" << G4VERSION_NUMBER
+                   << ") is too old to support the fast simulation Flush() "
+                      "interface");
+    CELER_EXPECT(params);
+    CELER_EXPECT(local);
+}
+
+FastSimulationOffload::FastSimulationOffload(G4String const& name,
```

Comment:
```suggestion
//---------------------------------------------------------------------------//
/*!
 * Construct a model for a specific volume of space (an "envelope").
 *
 * If the envelope is \c nullptr, the model is applicable globally.
 * 
 */
 FastSimulationOffload::FastSimulationOffload(G4String const& name,
```

Reply:
Done, but with clarification that the envelope pointer cannot be `nullptr` (Could we use/add `not_null` or something similar for cases like this?).